src/lug/Graphics/Vulkan/Gui.cpp

```diff
@@ -0,0 +1,876 @@
+                }
+            };
+            frameData.commandBuffer.setScissor({scissor});
+
+            const API::CommandBuffer::CmdDrawIndexed cmdDrawIndexed {
+                /* cmdDrawIndexed.indexCount */ pcmd->ElemCount,
+                /* cmdDrawIndexed.instanceCount */ 1,
+                /* cmdDrawIndexed.firstIndex */ indexCount,
+                /* cmdDrawIndexed.vertexOffset */ vertexCount,
+                /* cmdDrawIndexed.firstInstance */ 0
+            };
+            frameData.commandBuffer.drawIndexed(cmdDrawIndexed);
+
+            indexCount += pcmd->ElemCount;
+        }
+        vertexCount += cmd_list->VtxBuffer.Size;
+    }
+
+    frameData.commandBuffer.endRenderPass();
+    frameData.commandBuffer.end();
+
+    std::vector<VkPipelineStageFlags> waitDstStageMasks(waitSemaphores.size(), VK_PIPELINE_STAGE_TOP_OF_PIPE_BIT);
+    if (graphicsQueue->submit(frameData.commandBuffer, { static_cast<VkSemaphore>(frameData.semaphore) }, waitSemaphores, waitDstStageMasks, static_cast<VkFence>(frameData.fence)) == false) {
+        LUG_LOG.error("GUI: Can't submit commandBuffer");
+        return false;
+    }
+
+    return true;
+}
+
+void Gui::processEvents(lug::Window::Event event) {
+    ImGuiIO& io = ImGui::GetIO();
+    switch (event.type) {
+    case (lug::Window::Event::Type::KeyPressed):
+    case (lug::Window::Event::Type::KeyReleased):
+        io.KeysDown[static_cast<int>(event.key.code)] = (event.type == lug::Window::Event::Type::KeyPressed) ? true : false;
+
+        io.KeyCtrl = static_cast<int>(event.key.ctrl);
+        io.KeyShift = static_cast<int>(event.key.shift);
+        io.KeyAlt = static_cast<int>(event.key.alt);
+        io.KeySuper = static_cast<int>(event.key.system);
+        break;
+    case (lug::Window::Event::Type::CharEntered):
+        if (event.character.val > 0 && event.character.val < 0x10000) {
+            io.AddInputCharacter(static_cast<unsigned short>(event.character.val));
+        }
+        break;
+    default:
+        break;
+    }
+}
+
+const Vulkan::API::Semaphore& Gui::getGuiSemaphore(uint32_t currentImageIndex) const {
+    return _framesData[currentImageIndex].semaphore;
+}
+
+void Gui::updateBuffers(uint32_t currentImageIndex) {
+    ImDrawData* imDrawData = ImGui::GetDrawData();
+    if (!imDrawData) {
+        return;
+    }
+    API::Device &device = _renderer.getDevice();
+    FrameData& frameData = _framesData[currentImageIndex];
+
+    // Note: Alignment is done inside buffer creation
+    VkDeviceSize vertexBufferSize = imDrawData->TotalVtxCount * sizeof(ImDrawVert);
+    VkDeviceSize indexBufferSize = imDrawData->TotalIdxCount * sizeof(ImDrawIdx);
+
+    // Update buffers only if vertex or index count has been changed compared to current buffer size
+    {
+        // Vertex buffer
+        if ((static_cast<VkBuffer>(frameData.vertexBuffer) == VK_NULL_HANDLE) || (static_cast<int>(frameData.vertexBuffer.getRequirements().size) != imDrawData->TotalVtxCount)) {
```

Comment:
You are comparing a number of vertex and a size of buffer!
And if the size required is lesser, re-use the buffer.